src/python/pants/notes/2.17.x.md

```diff
@@ -1,5 +1,55 @@
 # 2.17.x Release Series
 
+## What's New
+
+### Overall
+
+The [`scie-pants` launcher binary](https://github.com/pantsbuild/scie-pants) is now the recommended way to run Pants.
+Use of a `./pants` shell script in a repository is deprecated. Among other benefits, the `scie-pants` launcher is able
+to automatically download and maintain a Python distribution for running Pants. It also allows you to run `pants`
+from any directory within a repository and not just the build root. See the [Pants  installation
+instructions](https://www.pantsbuild.org/docs/installation) for additional information on how to migrate to `scie-pants.`
+
+### Backends
+
+#### Python
+
+Python dependencies are now analyzed via an intrinsic rule implemented in Rust for increased performance
+over the prior Python implementation of the rule. See [#18854](https://github.com/pantsbuild/pants/pull/18854) for
+discusion of the performance improvements.
+
+Exported mutable virtualenvs can now include [PEP-660](https://peps.python.org/pep-0660/) editable installs of `python_distribution` targets.
+To enable this feature for a resolve, add that resolve's name to the `[export].py_editables_in_resolves` list in `pants.toml`.
+See [#18639](https://github.com/pantsbuild/pants/pull/18639) for details on Pants' PEP-660 implementation.
+
+The `python_awslambda` or `python_google_cloud_function` targets now
+support a new 'zip' layout, as recommended by the cloud vendors. This
+layout gives smaller packages and faster cold starts than the
+existing Lambdex layout, and will become the default in 2.18. See
+the docs [for `python_awslambda`](https://www.pantsbuild.org/v2.17/docs/awslambda-python#https://www.pantsbuild.org/v2.17/docs/awslambda-python#migrating-from-pants-216-and-earlier) 
+and [for `python_google_cloud_function`](https://www.pantsbuild.org/v2.17/docs/google-cloud-function-python#migrating-from-pants-216-and-earlier).
+
+#### New: Javascript (experimental!)
+
+Pants now has _very early_ experimental support for Javascript thanks to ongoing efforts by Tobias Nilsson and
+Theo Ribeiro. Please note that the Javascript backend is still incomplete, is being actively developed, and _probably
+has critical bugs_. Despite that, the Pants maintainers would appreciate any feedback from the community to help
+guide our development efforts.
+
+Enable the `pants.backend.experimental.javascript` backend to try out this support. Please file issues at
+https://github.com/pantsbuild/pants/issues/new/choose for any issues encountered.
```

Comment:
```suggestion
Pants now has _very early_ experimental support for Javascript thanks to ongoing efforts by Tobias Nilsson and
Theo Ribeiro. Please note that the Javascript backend is still incomplete, is being actively developed, and _probably
has critical bugs_. Despite that, the Pants maintainers would appreciate any feedback from the community to help
guide our development efforts.

Supported goals are:
* `test`: Letting you run tests via runners installed with the package manager of your choice.

* `package` goal, either to run a customized `package.json` script that produces an artifact via `node_build_script`, or to pack a tarball for npm-registry publication via the `npm_distribution` target.

* `tailor`: Generates build file targets for `*.js`, `*.test.js` and `package.json` files.

* `generate-lockfile`: Creates the lockfile in the format matching a projects package manager.

The backend supports all package managers provided by `corepack`. `yarn@v2` and [PlugNPlay](https://yarnpkg.com/features/pnp) is not supported.

Enable the `pants.backend.experimental.javascript` backend to try out this support. Please file issues at
https://github.com/pantsbuild/pants/issues/new/choose for any issues encountered, and follow along [the stabilization ticket on github](https://github.com/pantsbuild/pants/issues/19240).
```

Not sure how much prose should go here, but since there's no documentation (:see_no_evil:) maybe a list of what is possible helps encourage participation?